interfaces/cython/cantera/reaction.pyx

```diff
@@ -795,6 +795,44 @@ cdef class Reaction:
             if products:
                 self.products = products
 
+    def __init__(self, reactants=None, products=None, *, equation=None, rate=None,
+                 Kinetics kinetics=None, init=True, legacy=False, **kwargs):
+
+        if legacy:
+            return
+
+        if reactants and products and not equation:
+            equation = self.equation
+
+        if init and equation and kinetics:
+            rxn_type = self._reaction_type
+            spec = {"equation": equation, "type": rxn_type}
+            if isinstance(rate, dict):
+                # Arrhenius-like rates
+                spec["rate-constant"] = rate
+                rate = None
+            elif isinstance(rate, ReactionRate):
+                spec["type"] = rate.type
+
+            self._reaction = CxxNewReaction(dict_to_anymap(spec),
+                                            deref(kinetics.kinetics))
+            self.reaction = self._reaction.get()
+
+            if isinstance(rate, ReactionRate):
+                self.rate = rate
+                rate = None
+            elif rate is not None:
+                try:
+                    if isinstance(rate[0][1], Arrhenius):
+                        self.rate = PlogRate(rate)
+                        rate = None
+                except (IndexError, TypeError, KeyError):
+                    pass
```

Comment:
with the exception of a dictionary (see comment above) and the potential exception of a (YAML-compatible) string, I believe all other options should be removed. For `PlogRate`, it may make sense to convert the output from the nested list to a dictionary (an idea that I played with, but ultimately dropped for reasons of back-ward compatibility).